cmd/snap/complete.go

```diff
@@ -439,17 +439,25 @@ func (s serviceName) Complete(match string) []flags.Completion {
 		return nil
 	}
 
-	snaps := map[string]bool{}
+	snaps := map[string]int{}
 	var ret []flags.Completion
 	for _, app := range apps {
 		if !app.IsService() {
 			continue
 		}
-		if !snaps[app.Snap] {
-			snaps[app.Snap] = true
-			ret = append(ret, flags.Completion{Item: app.Snap})
+		name := snap.JoinSnapApp(app.Snap, app.Name)
+		if !strings.HasPrefix(name, match) {
+			continue
+		}
+		ret = append(ret, flags.Completion{Item: name})
+		if len(match) <= len(app.Snap) {
+			snaps[app.Snap]++
+		}
+	}
+	for snap, n := range snaps {
+		if n > 1 {
```

Comment:
Isn't this check unnecessary? It seems that the snap will only turn up in `snaps` if there is a partial match on the snap name.

Reply:
I'm having it complete straight to the service when there's only one of them

Reply:
Ahh ok, that's fine then.